be/src/olap/tablet_schema_cache.h

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <gen_cpp/olap_file.pb.h>
+
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+#include "olap/tablet_schema.h"
+
+namespace doris {
+
+class TabletSchemaCache {
+public:
+    static void create_global_schema_cache() {
+        DCHECK(_s_instance == nullptr);
+        static TabletSchemaCache instance;
+        _s_instance = &instance;
+    }
+
+    static TabletSchemaCache* instance() { return _s_instance; }
+
+    std::shared_ptr<TabletSchema> insert(const std::string& key) {
+        std::lock_guard guard(_mtx);
```

Comment:
Consider using `parallel-hashmap`, which may have better performance.

And, it has `lazy_emplace_l`, `if_contains` to handle concurrency, which can avoid using locks manually. This looks more concise, although less well understood.

Be careful not to use `try_emplace_l`, this behavior is not as expected https://github.com/greg7mdp/parallel-hashmap/issues/161